### PR TITLE
Remove support for deprecated UpstreamCA

### PR DESCRIPTION
**Pull request by azdagron**
Removes support for the UpstreamCA plugin, which was deprecated in favor of the UpstreamAuthority plugin in 0.10.0.

Also removes the server catalog unit-test which only tested deprecation behavior for UpstreamCA. A new unit-test that covers the catalog behavior will be added separately.

### CODE REVIEW

File: examples/plugins/upstreamauthority/plugin.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+	"github.com/hashicorp/hcl"
+	"github.com/spiffe/spire/pkg/common/catalog"
+	"github.com/spiffe/spire/pkg/server/plugin/upstreamauthority"
+	"github.com/spiffe/spire/proto/spire/common/plugin"
+	"github.com/zeebo/errs"
+)
+
+const (
+	// TODO: Replace with your plugin name. This will be used by the catalog
+	// to identify your plugin. Plugin names don't usually contain the plugin
```
Comment (amartinezfayo) [inline examples/plugins/upstreamauthority/plugin.go:17]:
```suggestion
	// to identify your plugin. Plugin names don't usually contain the plugin type
```

File: examples/plugins/upstreamauthority/plugin.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+// Publishes a JWT signing key upstream and responds with the upstream JWT
+// keys. If supported by the implementation, subsequent responses on the
+// stream contain upstream JWT key updates, otherwise the RPC is completed
+// after sending the initial response.
+//
+// This RPC is optional and will return NotImplemented if unsupported.
+//
+// Implementation note:
+// The stream should be kept open open in the face of transient errors
```
Comment (amartinezfayo) [inline examples/plugins/upstreamauthority/plugin.go:104]:
```suggestion
// The stream should be kept open in the face of transient errors
```

**Comment by azdagron** [comment]
I fixed them up as part of this PR but rolled the change back. The two configs are no longer in use (they were part of an HA playground that is no longer around). I was about to open up a PR to remove them.

File: examples/plugins/upstreamauthority/plugin.go (hunk 1 above)
Comment (azdagron) [reply examples/plugins/upstreamauthority/plugin.go]: :+1:

File: examples/plugins/upstreamauthority/plugin.go (hunk 2 above)
Comment (azdagron) [reply examples/plugins/upstreamauthority/plugin.go]: This is also in the upstreamauthority.proto. I'll fix both.

**Pull request merged**